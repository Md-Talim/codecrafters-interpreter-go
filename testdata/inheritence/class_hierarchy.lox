// Test Case 1
class Doughnut {}

class BostonCream < Doughnut {}

print Doughnut();
print BostonCream();

// Test Case 2
{
  class A {}

  // B is a subclass of A
  class B < A {}

  // C is also a subclass of A
  class C < A {}

  print A();
  print B();
  print C();
}

// Test Case 3
class Vehicle {}

// Car is a subclass of Vehicle
class Car < Vehicle {}

// Sedan is a subclass of Car
class Sedan < Car {}

print Vehicle();
print Car();
print Sedan();

{
  // Truck is a subclass of Vehicle
  class Truck < Vehicle {}
  print Truck();
}