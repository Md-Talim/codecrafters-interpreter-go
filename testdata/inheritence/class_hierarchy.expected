Doughnut instance
BostonCream instance
A instance
B instance
C instance
Vehicle instance
Car instance
Sedan instance
Truck instance